src/lib.rs

```diff
@@ -4158,7 +4158,7 @@ fn is_disabled() -> bool {
             // Not set? Not disabled.
             None => false,
             // Respect `CC_FORCE_DISABLE=0` and some simple synonyms.
-            Some(v) if &*v != "0" && &*v != "false" && &*v != "no" => false,
+            Some(v) if &*v == "0" || &*v == "false" || &*v == "no" => false,
```

Comment:
As a quick fix this looks fine but is there a better way to write this? It does feel a bit like symbol soup and obviously the logic error was missed the first time so it'd be great if it could be made clearer.

Reply:
I tried `matches!(v, "0" | "false" | "no")`, but that doesn't work because this is an `OsString`. Can't think of anything else.

Reply:
Hm, an `as_deref()` would help avoid the `&*` at least by giving us an `OsStr`. But maybe we can do a bit better at the cost of some verbosity?

```rust
fn compute_is_disabled() -> bool {
    match std::env::var_os("CC_FORCE_DISABLE") {
        // Not set? Not disabled.
        None => false,
        Some(cc_force_disable) => match cc_force_disable.to_str() {
            // Respect `CC_FORCE_DISABLE=0` and some simple synonyms.
            Some("0" | "false" | "no") => false,
            // Otherwise, we're disabled. This intentionally includes `CC_FORCE_DISABLE=""`
            _ => true,
        },
    }
}
```

Reply:
I'll leave that decision to @NobodyXu.

Reply:
I think it looks ok now, will merge and release